app/api/services/informationextraction/getFiles.ts

```diff
@@ -71,7 +75,17 @@ async function getFilesForTraining(templates: ObjectIdSchema[], property: string
     { limit: MAX_TRAINING_FILES_NUMBER }
   )) as (FileType & FileEnforcedNotUndefined)[];
 
-  return getFilesWithAggregations(files);
+  const filesWithEntityValue = files.map(file => {
+    const entity = entities.find(e => e.sharedId === file.entity);
```

Comment:
This goes through every entity and every related file, which can be a lot, so this is potentially slow (find inside map, running time scales with (number_of_entites*number_of_files). Consider pre-processing the entities, mapping the values, so here instead of the find it is just picking something from a map or object.
The indexObject function in shared/dataUtils is for these cases.